src/storage/data_table.cpp

```diff
@@ -11,28 +11,26 @@
 
 namespace terrier::storage {
 
-DataTable::DataTable(const common::ManagedPointer<BlockStore> store, const BlockLayout &layout,
+DataTable::DataTable(common::ManagedPointer<BlockStore> store, const BlockLayout &layout,
                      const layout_version_t layout_version)
-    : block_store_(store), layout_version_(layout_version), accessor_(layout) {
+    : accessor_(layout), block_store_(store), layout_version_(layout_version) {
   TERRIER_ASSERT(layout.AttrSize(VERSION_POINTER_COLUMN_ID) == 8,
                  "First column must have size 8 for the version chain.");
   TERRIER_ASSERT(layout.NumColumns() > NUM_RESERVED_COLUMNS,
                  "First column is reserved for version info, second column is reserved for logical delete.");
-  if (block_store_ != nullptr) {
-    RawBlock *new_block = NewBlock();
-    // insert block
-    blocks_.push_back(new_block);
+  if (store != DISABLED) {
+    blocks_.push_back(NewBlock());
+    blocks_size_++;
   }
-  insertion_head_ = 0;
 }
 
 DataTable::~DataTable() {
-  common::SpinLatch::ScopedSpinLatch guard(&blocks_latch_);
-  for (RawBlock *block : blocks_) {
+  common::SharedLatch::ScopedExclusiveLatch latch(&blocks_latch_);
+  for (auto block : blocks_) {
     StorageUtil::DeallocateVarlens(block, accessor_);
     for (col_id_t i : accessor_.GetBlockLayout().Varlens())
       accessor_.GetArrowBlockMetadata(block).GetColumnInfo(accessor_.GetBlockLayout(), i).Deallocate();
-    block_store_->Release(block);
+    block_store_.operator->()->Release(block);
```

Comment:
Why did you change this to explicitly invoke the arrow operator rather than how it was?

Reply:
done